guides/source/active_record_migrations.md

```diff
@@ -927,43 +1172,66 @@ number associated with the previous migration you can run:
 $ bin/rails db:rollback
 ```
 
-This will rollback the latest migration, either by reverting the `change`
-method or by running the `down` method. If you need to undo
-several migrations you can provide a `STEP` parameter:
+This will rollback the latest migration, either by reverting the `change` method
+or by running the `down` method. If you need to undo several migrations you can
+provide a `STEP` parameter:
 
 ```bash
 $ bin/rails db:rollback STEP=3
 ```
 
 The last 3 migrations will be reverted.
 
-The `db:migrate:redo` command is a shortcut for doing a rollback and then migrating
-back up again. As with the `db:rollback` command, you can use the `STEP` parameter
-if you need to go more than one version back, for example:
+In some cases where you modify a local migration and would like to rollback that
+specific migration before migrating back up again, you can use the
+`db:migrate:redo` command. As with the `db:rollback` command, you can use the
+`STEP` parameter if you need to go more than one version back, for example:
 
 ```bash
 $ bin/rails db:migrate:redo STEP=3
 ```
 
-Neither of these rails commands do anything you could not do with `db:migrate`. They
-are there for convenience, since you do not need to explicitly specify the
-version to migrate to.
+NOTE: You could get the same result using `db:migrate`. However, these are there
+for convenience so that you do not need to explicitly specify the version to
+migrate to.
+
+#### Transactions
+
+In databases that support transactions with statements that change the schema,
+each migration is wrapped in a transaction.
+
+INFO: A transaction ensures that if a migration fails partway through, any
+changes that were successfully applied are rolled back, maintaining database
+consistency. This means that either all operations within the transaction are
+executed successfully, or none of them are, preventing the database from being
+left in an inconsistent state if an error occurs during the transaction.
+
+If the database does not support transactions with statements that change the
```

Comment:
Postgres calls this "Transactional DDL" and the major operations are all supported, outside of creating indexes concurrently. 

Here's a writeup: https://thenewstack.io/why-and-when-you-need-transactional-ddl-in-your-database/

It might be worth describing what we can expect for the other supported database adapters.

If that's too much scope for this section, I'd be happy to contribute a writeup externally that could possibly be included as an appendix or be linked to.

This can be a confusing area for developers as it relates to transactions, DDL, up/down migrations, and inside/outside transactions (concurrently).

Reply:
Maybe that belongs in the Postgres guide?

Reply:
The API docs also call these [DDL transactions](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Migration.html#class-ActiveRecord::Migration-label-Transactional+Migrations). It also helps in understanding the `disable_ddl_transaction!` method mentioned later on.
```suggestion
If the database does not support DDL transactions with statements that change the
```